### PR TITLE
NAS-3716 - Generate new client for versioned api

### PR DESCRIPTION
**Pull request by hackerstanislav**
JIRA: NAS-3716

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8141/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9963/ : FAILURE in 2m 02s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8143/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9965/ : FAILURE in 2m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8144/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9966/ : FAILURE in 20m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8146/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9968/ : FAILURE in 22m 18s

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts [hunk 1]
```diff
@@ -170,7 +170,8 @@ class TigerWorkspaceElementsQuery implements IElementsQuery {
                     items: elements.map(
                         (element): IAttributeElement => ({
                             title: element.title,
-                            uri: element.primaryTitle,
+                            //TODO sha: Primary label is optional now
+                            uri: element.primaryTitle!,
```
Comment (sadam21) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:174]: It should stay nullable but required, shouldn't it?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8204/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10047/ : FAILURE in 17m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/8212/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10053/ : SUCCESS in 17m 33s

File: libs/sdk-model/src/objRef/index.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export type ObjectType =
     | "tag"
     | "insight"
     | "variable"
+    | "prompt"
```
Comment (no23reason) [inline libs/sdk-model/src/objRef/index.ts:48]: I am not sure we want to expose/support this. Is this necessary?

Comment (hackerstanislav) [reply libs/sdk-model/src/objRef/index.ts:48]: Probably not using now, can i delete it from tests or also from enum?

File: libs/api-client-tiger/src/gd-tiger-model/VisualizationObjectModelV1.ts [hunk 3]
```diff
@@ -67,9 +67,9 @@ export namespace VisualizationObjectModelV1 {
      * @deprecated use {@link GrandTotal} instead
      */
     export interface ITotalItem {
-        measureIdentifier: LocalIdentifier;
+        measureIdentifier: AfmIdentifier;
```
Comment (no23reason) [inline libs/api-client-tiger/src/gd-tiger-model/VisualizationObjectModelV1.ts:70]: Should likely be `AfmLocalIdentifier`, right?

Comment (hackerstanislav) [reply libs/api-client-tiger/src/gd-tiger-model/VisualizationObjectModelV1.ts:70]: Yep, updating in next commit 👍

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2566

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2353/ : SUCCESS in 1m 43s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2647/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1465/ : SUCCESS in 16m 03s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/10061/ : SUCCESS in 16m 08s

**Pull request merged**